api/deploy.go

```diff
@@ -88,6 +88,9 @@ func deploy(w http.ResponseWriter, r *http.Request, t auth.Token) (err error) {
 	if err != nil {
 		return &tsuruErrors.HTTP{Code: http.StatusNotFound, Message: err.Error()}
 	}
+	if (instance.GetPlatform() == "") && (origin != "image") {
+		return &tsuruErrors.HTTP{Code: http.StatusUnauthorized, Message: "can't deploy app without platform, if it's not an image"}
```

Comment:
I think an http.StatusBadRequest would fit better here. Also, perhaps we should allow `rollback` too (cc @cezarsa)? Maybe we could move this logic to the `app.ValidateOrigin` func.